src/common/pca_9685.cc

```diff
@@ -0,0 +1,138 @@
+#ifdef __linux__
+#include "common/pca_9685.h"
+
+// BoB robotics include
+#include "plog/Log.h"
+
+namespace
+{
+union PWM
+{
+    uint16_t ushort[2];
+    uint8_t byte[4];
+};
+}
+//----------------------------------------------------------------------------
+// BoBRobotics::PCA9685
+//----------------------------------------------------------------------------
+namespace BoBRobotics
+{
+PCA9685::PCA9685(int slaveAddress, const char *path, units::frequency::hertz_t referenceClockSpeed)
+:   m_ReferenceClockSpeed(referenceClockSpeed)
+{
+    // Setup I2C device
+    m_Device.setup(path, slaveAddress);
+    
+    reset();
+}
+//----------------------------------------------------------------------------
+void PCA9685::setDutyCycle(Channel channel, float dutyCycle)
+{
+    // Clamp duty cycle
+    dutyCycle = std::min(1.0f, std::max(0.0f, dutyCycle));
```

Comment:
Elsewhere in BoB robotics I've generally used error checks rather than silently capping values. The danger is that your code ends up passing some nonsense value in and you never notice.

How about:
```c++
BOB_ASSERT(dutyCycle >= 0.f && dutyCycle <= 1.f);
```